v2/pkg/operator/local_stored_collector.go

```diff
@@ -220,17 +241,36 @@ func (o LocalStorageCollector) prepareM2DCopyBatch(log clog.PluggableLoggerInter
 		for _, img := range relatedImgs {
 			var src string
 			var dest string
-			imgSpec, err := image.ParseRef(img.Image)
-			if err != nil {
-				o.Log.Error("%s", err.Error())
-				return nil, err
-			}
-			src = imgSpec.ReferenceWithTransport
-
-			if imgSpec.IsImageByDigest() {
-				dest = dockerProtocol + strings.Join([]string{o.LocalStorageFQDN, imgSpec.PathComponent + ":" + imgSpec.Digest[:hashTruncLen]}, "/")
+			if strings.HasPrefix(img.Image, ociProtocol) {
+				transportAndPath := strings.Split(img.Image, "://")
+				src = ociProtocolTrimmed + transportAndPath[1]
+				if len(transportAndPath) == 2 {
+					// generate a random hex string to use as a tag
+					var rnd = rand.New(rand.NewSource(time.Now().UnixNano()))
+					hex := strconv.FormatUint(rnd.Uint64(), 16)
+					var hexTag string
+					if len(hex) > hashTruncLen {
+						hexTag = hex[:hashTruncLen]
+					} else {
+						hexTag = hex
+					}
+					dest = dockerProtocol + strings.Join([]string{o.LocalStorageFQDN, transportAndPath[1] + ":" + hexTag}, "/")
+				} else {
+					dest = dockerProtocol + strings.Join([]string{o.LocalStorageFQDN, img.Image}, "/")
```

Comment:
Are we sure this is going to be accepted by the registry? ie no tag nor digest? 
Shouldn't we use the digest from the manifest instead here?
Maybe also (at least for oci), check if `targetTag` is provided: I bet the users of the oci feature would be looking for this.

Reply:
I believe we should use the digest from the manifest on line 257 (instead of a random one) and here on line 259 we should try to find the tag (checking if the image contains a tag or if it is present in the targetTag field).